go/glome/glome_test.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package glome
+
+import (
+	"bytes"
+	"encoding/hex"
+	"testing"
+)
+
+func handle(e error, t *testing.T) {
+	if e != nil {
+		t.Fatalf("Unexpected Error: " + e.Error())
+	}
+}
+
+// Stores test vectors from protocol reference. Each variable is named after
+// a row of the test table. For the purpose of testing, we consider that user A
+// is always the one that sends the message (therefore, we change the role of A
+// and B in Vector #2)
+type testVector struct {
+	kap     []byte //kap = K_a'(k sub a *p*rime)
+	ka      []byte
+	kbp     []byte
+	kb      []byte
+	counter uint8
+	msg     []byte
+	ks      []byte
+	tag     []byte
+}
+
+func (tv *testVector) Dialogs(t *testing.T) (*Dialog, *Dialog) {
+	aPriv, err := PrivateKeyFromSlice(tv.kap)
+	handle(err, t)
+	aPub, err := PublicKeyFromSlice(tv.ka)
+	handle(err, t)
+
+	bPriv, err := PrivateKeyFromSlice(tv.kbp)
+	handle(err, t)
+	bPub, err := PublicKeyFromSlice(tv.kb)
+	handle(err, t)
+
+	sending, err := aPriv.Exchange(bPub)
+	handle(err, t)
+	receiving, err := bPriv.Exchange(aPub)
+	handle(err, t)
+
+	return sending, receiving
+}
+
+func decodeString(s string) []byte {
+	b, err := hex.DecodeString(s)
+	if err != nil {
+		return []byte{0} // We return a single byte.
```

Comment:
Why? If this is invalid input in a test, then panic here instead of passing along corrupted data.

Reply:
Fix.